xblock/test/test_core.py

```diff
@@ -939,48 +950,78 @@ def stub_resource_stream(self, module, name):
         return "!" + name + "!"
 
     @ddt.data(
-        "public/hey.js",
-        "public/sub/hey.js",
-        "public/js/vendor/jNotify.jQuery.min.js",
-        "public/something.foo",         # Unknown file extension is fine
-        "public/a/long/PATH/no-problem=here$123.ext",
-        "public/ℓιвяαяу.js",
+        u"public/hey.js",
+        u"public/sub/hey.js",
+        u"public/js/vendor/jNotify.jQuery.min.js",
+        u"public/something.foo",         # Unknown file extension is fine
+        u"public/a/long/PATH/no-problem=here$123.ext",
+        u"public/\N{SNOWMAN}.js",
     )
     def test_open_good_local_resource(self, uri):
         loadable = self.LoadableXBlock(None, scope_ids=None)
         with patch('pkg_resources.resource_stream', self.stub_resource_stream):
             assert loadable.open_local_resource(uri) == "!" + uri + "!"
+            assert loadable.open_local_resource(uri.encode('utf-8')) == "!" + uri + "!"
 
     @ddt.data(
-        "public/../secret.js",
-        "public/.git/secret.js",
-        "static/secret.js",
-        "../public/no-no.bad",
-        "image.png",
-        ".git/secret.js",
-        "static/ℓιвяαяу.js",
+        u"public/hey.js".encode('utf-8'),
```

Comment:
In part because we would have had to represent the multi-byte values as their individual bytes.  I don't remember if there was a reason to do them for the ascii-equivalent bytestrings.